### ISSUE TITLE
Proxying Loris

### ISSUE THREAD

**Issue by lfarrell** [loris-imageserver/loris#220]
Does loris support something like this: https://groups.google.com/forum/?hl=en#!topic/iiif-discuss/207JToiwl_0? We'd like to proxy all requests for info.json and image tiles through our larger application. 

Thanks,

Dean

**Comment by jpstroop** [comment]
Dean
It's come up before, but I think the person who asked wound up solving it w/ ports (they were using Varnish or Squid or something like that, on the same server). It terms of natively changing the base URI (in IIIF terms) no. Loris does not have that feature. 

That said, it would be pretty trivial to implement: the base URI is calculated in from the request [here](https://github.com/loris-imageserver/loris/blob/development/loris/webapp.py#L359-L362), but an override option could be added to the config file [read into the webapp here](https://github.com/loris-imageserver/loris/blob/development/loris/webapp.py#L192-L225) and used instead if it is set. As far as I can remember, that value is just passed around and just used in the `info.json` serialization.

**Comment by lfarrell** [comment]
Thanks, I'll take a look. At the moment I've modified the leaflet-IIIF plugin to work how we'd like: 
So passing in this: 

``` JavaScript
 var iiifLayers = {'img': L.tileLayer.iiif('jp2Metadata/' + this.options.url + '/IMAGE_JP2000',
         { tileProxyPath: '/jp2Region/' + this.options.url + '/IMAGE_JP2000', mapSelector: '#jp2_viewer'})
  };
```

Outputs the below, while still passing everything through proxies

``` JavaScript
{"profile": ["http://iiif.io/api/image/2/level2.json", {"supports": ["canonicalLinkHeader", "profileLinkHeader"
, "mirroring", "rotationArbitrary", "sizeAboveFull"], "qualities": ["default", "bitonal", "gray", "color"
], "formats": ["jpg", "png", "gif", "webp"]}], "tiles": [{"width": 256, "scaleFactors": [1, 2, 4, 8,
 16, 32]}], "protocol": "http://iiif.io/api/image", "sizes": [{"width": 119, "height": 147}, {"width"
: 238, "height": 294}, {"width": 475, "height": 588}, {"width": 949, "height": 1176}, {"width": 1898
, "height": 2351}, {"width": 3795, "height": 4701}], "height": 4701, "width": 3795, "@context": "http
://iiif.io/api/image/2/context.json", "@id": "https://cdr-qa.lib.unc.edu/loris/uuid%3Adc598587-47f6-41a9-8835-8378a8363a5b"
}
```

Don't really want to maintain our own fork of Leaflet-IIIF, however.

**Comment by jpstroop** [comment]
Yeah, if you did it I'd happily merge :wink:

**Comment by jpstroop** [comment]
Closed by #221
